### ISSUE TITLE
Filter stations with insufficient label history before training

### ISSUE THREAD

**Issue by edwinsteele** [edwinsteele/fuel-price-signal#29]
## What

Add a minimum-row filter when loading \`data/labels.csv\` for training. Two classes of problem stations identified during label diagnostic review:

1. **Too-new stations** (7 stations, 21–53 rows): station codes 20619, 20626, 20638, 20649, 20652, 20653, 20656. All opened recently, haven't survived a full price cycle, positive rate = 0%. Dead weight in training.

2. **Data-gap distortion** (at least 2 stations confirmed): station 20528 (Speedway William Street, Granville) positive rate 0.84, median price 116.9c; station 20133 (Metro Condell Park West) positive rate 0.72, median price 143.7c. Anomalously low medians vs Sydney-wide 2021–2026 prices suggest the stations went offline during high-price years. Their rolling P33 was computed against a cheap-only history, causing both label conditions to fire almost constantly.

## Where the filter lives

Training load time, not in \`labels.py\`. \`labels.py\` produces correct labels — this is a modelling data-quality concern, not a label correctness concern.

Suggested implementation in the training loader:
\`\`\`python
min_rows = 365  # roughly one year of labeled data
counts = df.groupby("station_code")["label"].count()
eligible = counts[counts >= min_rows].index
df = df[df["station_code"].isin(eligible)]
\`\`\`

## Acceptance criteria

- [ ] Minimum row threshold (365) applied when assembling the training dataset
- [ ] Threshold is a named constant or parameter, not a magic number
- [ ] Stations 20528 and 20133 investigated — confirm whether data gap is the cause; if so, document in a comment or config exclusion list
- [ ] Test: a station with < 365 rows is excluded from the assembled training set
